### DISCUSSION TITLE
Package rules by group

### DISCUSSION THREAD

**Discussion by DerYeger** [renovatebot/renovate#11421] [Request Help]
Currently, I'm using the following configuration to prevent Renovate from updating the Jest monorepo to v27.

```json
"packageRules": [
  {
    "matchPackageNames": ["babel-jest", "jest", "ts-jest"],
    "allowedVersions": "<27"
  }
]
```

I noticed that all of these dependencies are already grouped together as `jest monorepo`. Is it possible to do something along the lines of the following snippet?

```json
"packageRules": [
  {
    "groupName": "jest monorepo",
    "allowedVersions": "<27"
  }
]
```

**Comment by rarkins** [comment]
Try

```
"packageRules": [
  {
    "extends": ["monorepo:jest"],
    "allowedVersions": "<27"
  }
]
```

But if you're using GitHub or GitLab then instead I'd recommend:

```
"packageRules": [
  {
    "extends": ["monorepo:jest"],
    "matchUpdateTypes": ["major"],
    "dependencyDashboardApproval": true
  }
]
```

Otherwise you lose visibility into this upgrade plus need to make a config change if/when you decide to upgrade. I tend to set all major upgrades to approval like this.

**Reply by DerYeger** [reply]
Thank you! I tried the second variant (see https://github.com/DerYeger/vue-masonry-wall/issues/4 and https://github.com/DerYeger/vue-masonry-wall/blob/master/renovate.json), but this caused Renovate to reopen the PR instead of requesting approval. Is this intended behavior?

**Reply by rarkins** [reply]
It's not, but it also doesn't seem to be the exact rule I suggested.

**Reply by DerYeger** [reply]
> It's not, but it also doesn't seem to be the exact rule I suggested.

Yes, I had to include `@types/jest` as well. I thought it wouldn't be a problem, since it checks if either `match...` key applies.

**Reply by rarkins** [reply]
Incorrect. It's an AND match, not OR

**Reply by DerYeger** [reply]
Ahh, I see. The documentation states `"However you can mix together both matchPackageNames and matchPackagePatterns in the same package rule and the rule will be applied if either match."` at https://docs.renovatebot.com/configuration-options/#packagerules/ which misled me.

**Comment by viceice** [comment]
Nope, you need at least one `match...` key in a package rule

**Reply by DerYeger** [reply]
> Nope, you need at least one `match...` key in a package rule

I see, so there's nothing like `matchGroupName` yet? Thank you for the fast help!
